astroplan/core.py

```diff
@@ -157,6 +157,28 @@ def __init__(self, location=None, timezone='UTC', name=None, latitude=None,
             raise TypeError('timezone keyword should be a string, or an '
                             'instance of datetime.tzinfo')
 
+    @classmethod
+    def at_site(cls, site_name, **kwargs):
+        """
+        Initialize an `~astroplan.core.Observer` object with a site name.
+
+        Parameters
+        ----------
+        site_name : str
+            Observatory name, must be resolvable with
+            `~astroplan.sites.get_site`
+
```

Comment:
You should add a mention either in the `Parameters` section or the text above that any additional keywords are passed into this class' constructor.  Maybe also mention in the docs that if you *don't* specifically give a `name`, `site_name` will be used as the name.